src/db.ts

```diff
@@ -62,91 +62,36 @@ const BeginTransactionMethodName = "/tigrisdata.v1.Tigris/BeginTransaction";
  * ```
  */
 export type TemplatedBranchName = { name: string; dynamicCreation: boolean };
-const NoBranch: TemplatedBranchName = { name: "", dynamicCreation: false };
+const DefaultBranch: TemplatedBranchName = { name: "main", dynamicCreation: false };
 
 /**
  * Tigris Database class to manage database branches, collections and execute
  * transactions.
  */
 export class DB {
-	private readonly _db: string;
-	private _branch: string;
+	private readonly _name: string;
+	private readonly _branchVar: TemplatedBranchName;
 	private readonly grpcClient: TigrisClient;
 	private readonly config: TigrisClientConfig;
 	private readonly schemaProcessor: DecoratedSchemaProcessor;
 	private readonly _metadataStorage: DecoratorMetaStorage;
-	private readonly _ready: Promise<this>;
 
 	/**
 	 * Create an instance of Tigris Database class.
 	 *
 	 * @example Recommended way to create instance using {@link TigrisClient.getDatabase}
 	 * ```
 	 * const client = new TigrisClient();
-	 * const db = await client.getDatabase();
-	 * ```
-	 *
-	 * @remarks
-	 * Highly recommend to use {@link TigrisClient.getDatabase} to create instance of this class and
-	 * not attempt to instantiate this class directly, it can have potential side effects
-	 * if database branch is not initialized properly.
-	 *
-	 * @privateRemarks
-	 * Object of this class depends on readiness state for proper initialization of database
-	 * and branch. To ensure object is ready to use:
-	 * ```
-	 * const instance = new DB(name, client, config);
-	 * const db: DB = await instance.ready;
-	 *
-	 * db.describe();
+	 * const db = client.getDatabase();
 	 * ```
 	 */
 	constructor(db: string, grpcClient: TigrisClient, config: TigrisClientConfig) {
-		this._db = db;
+		this._name = db;
 		this.grpcClient = grpcClient;
 		this.config = config;
 		this.schemaProcessor = DecoratedSchemaProcessor.Instance;
 		this._metadataStorage = getDecoratorMetaStorage();
-		this._branch = NoBranch.name;
-		this._ready = this.initializeDB();
-	}
-
-	/**
-	 * Initializes a database branch and returns DB object. A DB shouldn't be used
-	 * until it is initialized.
-	 *
-	 * Calls {@link describe()} to assert that the branch in use already exists. If not, and the
-	 * branch name needs to be generated dynamically (ex - `preview_${GIT_BRANCH}`) then try to
-	 * create that branch.
-	 *
-	 * @throws Error if branch doesn't exist and/or cannot be created
-	 * @private
-	 */
-	private async initializeDB(): Promise<this> {
```

Comment:
This method was initially called in constructor, decoupled it as a standalone call